src/RadixDlt.NetworkGateway.Abstractions/Model/PendingTransactionIntentLedgerStatus.cs

```diff
@@ -0,0 +1,136 @@
+ * Work or any use of the Work complies with any law or regulation in any territory where
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using System;
+
+namespace RadixDlt.NetworkGateway.Abstractions.Model;
+
+/// <summary>
+/// Any additional knowledge we have about the intent.
+/// </summary>
+public enum PendingTransactionIntentLedgerStatus
+{
+    /// <summary>
+    /// We know that the intent has been committed as a success.
+    /// </summary>
+    CommittedSuccess,
+
+    /// <summary>
+    /// We know that the intent has been committed as a failure.
+    /// </summary>
+    CommittedFailure,
+
+    /// <summary>
+    /// We know that the intent has been committed, but don't yet know the outcome until it's synced (Success/Failure).
```

Comment:
Open question: how can we tell something has been committed not seeing it in the committed TX stream? My guess is that this is going to be driven by the fact that we've seen given TX in mempool and then it disappeared? This does not sound like enough information to conclude that the TX got committed to me.

Reply:
No. This is because a Core API has told us that it's been committed at submission time, we just haven't synced it yet.